uefi-test-runner/src/proto/shell.rs

```diff
@@ -0,0 +1,40 @@
+use uefi::{prelude::*, proto::{shell::Shell, media::fs::SimpleFileSystem, device_path::{build::DevicePathBuilder, media::FilePath}}, table::boot::{SearchType, LoadImageSource}};
+
+pub fn test(bt: &BootServices) {
+    info!("Setting up for EFI Shell tests.");
+    let fs_handles = bt.locate_handle_buffer(SearchType::from_proto::<SimpleFileSystem>()).unwrap().handles();
+    for fs in fs_handles {
+        let mut sfs = bt.open_protocol_exclusive::<SimpleFileSystem>(*fs).expect("sfs to exist");
+        let root = (*sfs).open_volume().unwrap();
+        let device_path = root.device();
+    }
+
+    info!("Running EFI Shell tests.");
+    let handle = bt.find_handles::<Shell>()
+        .expect("system to have EFI Shell available");
```

Comment:
After correcting the problems enough that this can run with `cargo xtask run`, this `expect()` fails:

```
[PANIC]: panicked at 'system to have EFI Shell available: Error { status: NOT_FOUND, data: () }', uefi-test-runner/src/proto/shell.rs:23:10
```